File: aerox.Runtime.Widgets/Events/CursorUpEvent.cs
using aerox.Runtime.Math;
using aerox.Runtime.Widgets.Graphics;
using aerox.Runtime.Windows;

namespace aerox.Runtime.Widgets.Events;

public class CursorUpEvent : Event
{
    public Vector2<float> Position;
    public MouseButton Button;

    public CursorUpEvent(Surface surface,MouseButton button, Vector2<float> position) : base(surface)
    {
        Button = button;
        Position = position;
    }
}

File: modules/widgets/lib/aerox/widgets/event/CursorUpEvent.cpp
#include "aerox/widgets/event/CursorUpEvent.hpp"

namespace aerox::widgets
{
    CursorUpEvent::CursorUpEvent(const Shared<Surface>& inSurface, const window::ECursorButton inButton,
        const Vec2<float>& inPosition) : Event(inSurface), button(inButton), position(inPosition)
    {
        
    }
}
